### ISSUE TITLE
Unable to execute `bazel info` without a module file

### ISSUE THREAD

**Issue by shs96c** [bazelbuild/bazel#24239]
### Description of the bug:

See [the build logs](https://buildkite.com/bazel/rules-jvm-external-examples/builds/4407#0193073d-7cb4-43e8-bbf6-c8c1390b8b4f/116-128) from [this PR in rules_jvm_external](https://github.com/bazel-contrib/rules_jvm_external/pull/1279)

The execution of `bazel info` fails with:

```shell
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.NullPointerException: Cannot invoke "java.lang.Throwable.getMessage()" because "cause" is null
	at com.google.devtools.build.lib.analysis.config.InvalidConfigurationException.<init>(InvalidConfigurationException.java:53)
	at com.google.devtools.build.lib.skyframe.SkyframeExecutor.createBuildConfigurationKey(SkyframeExecutor.java:1813)
	at com.google.devtools.build.lib.skyframe.SkyframeExecutor.getConfiguration(SkyframeExecutor.java:1752)
	at com.google.devtools.build.lib.runtime.commands.InfoCommand.lambda$exec$0(InfoCommand.java:158)
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:181)
	at com.google.devtools.build.lib.runtime.commands.InfoCommand.exec(InfoCommand.java:215)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:664)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:244)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:573)
	at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$1(GrpcServerImpl.java:641)
	at io.grpc.Context$1.run(Context.java:566)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Traceback (most recent call last):
  File "bazelci.py", line 4493, in <module>
    sys.exit(main())
  File "bazelci.py", line 4471, in main
    bazel_version=task_config.get("bazel") or configs.get("bazel"),
  File "bazelci.py", line 1212, in execute_commands
    bazel_version = print_bazel_version_info(bazel_binary, platform)
  File "bazelci.py", line 1515, in print_bazel_version_info
    [bazel_binary] + common_startup_flags() + ["--nosystem_rc", "--nohome_rc", "info"]
  File "bazelci.py", line 2653, in execute_command
    subprocess.PIPE if capture_stderr else None
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['bazel', '--nosystem_rc', '--nohome_rc', 'info']' returned non-zero exit status 37.
```

### Which category does this issue belong to?

Core

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

1. Clone `rules_jvm_external` and checks out `d8cc6c07f2c1c61276f46e6f03b270db497e98fe`. 
2. `cd examples/simple`
3. `bazel info`

### Which operating system are you running Bazel on?

macOS and Linux

### What is the output of `bazel info release`?

7.4.0

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?

```text
https://github.com/bazel-contrib/rules_jvm_external.git
d8cc6c07f2c1c61276f46e6f03b270db497e98fe
```

**Comment by shs96c** [comment]
Digging a little deeper, the error was caused by skylib being referenced before skylib had been loaded in a workspace-based build.

**Comment by katre** [comment]
Yes, looks like a few places are creating an `InvalidConfigurationException` without checking for cycles, assuming there is a valid exception.

I'll add some checks for that.

**Comment by katre** [comment]
I think I have a better fix but I can't reproduce the issue, I'll put up a draft PR in a few minutes, would you be able to test it?

**Comment by katre** [comment]
Take a look at https://github.com/bazelbuild/bazel/pull/24241

**Comment by shs96c** [comment]
Works on my machine :) Thank you for the more complete fix!

**Comment by katre** [comment]
okay, I'll split it out and merge mine, and drop #24240.

**Comment by katre** [comment]
Okay, fix is submitted at head, @shs96c can you handle the cherrypick requests for 7.4.x and 8.0.0?

**Comment by fmeum** [comment]
@bazel-io fork 8.0.0

**Comment by fmeum** [comment]
@bazel-io fork 7.4.1

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.0.0 RC3](https://github.com/bazelbuild/bazel/releases/tag/8.0.0rc3). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.0.0rc3. Thanks!
